docs/design/automatic-updates.md

```diff
@@ -0,0 +1,38 @@
+
+# Storage Node automatic updates DesignDocument
+
+## Overview
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date. If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.
+
+## Goals
+
+Create an automatic update system into the Storage Node software so that node operators do not have to worry about updating their nodes.
+Keep the Storage Nodes on the network up to date with the latest software to improve stability and performance. 
```

Comment:
This doc doesn't contain rollouts.

Reply:
I don't know what that is

Reply:
https://en.wikipedia.org/wiki/Rolling_release

Reply:
Gradual rollouts meaning updating all the storage-nodes incrementally rather than all at once.

Reply:
We can't control when these nodes update without adding a new api call

Reply:
Yes, this means the design needs to change. We discussed the rollouts during all hands, so we should have notes about having the counter etc.

Reply:
added a comment about what we discussed on the PR top-level